src/services/auth.service.js

```diff
@@ -0,0 +1,47 @@
+const jwt = require("jsonwebtoken");
+const bcrypt = require("bcrypt");
+
+const userDao = require("../models/user.dao");
+
+const signUp = async (
+  name,
+  dateOfBirth,
+  phoneNumber,
+  gender,
+  address,
+  email,
+  password
+) => {
+  const user = await userDao.getUserByEmail(email);
+  if (user === undefined) {
+    const hashedPassword = await bcrypt.hash(password, 10);
+    await userDao.createUser(
+      name,
+      dateOfBirth,
+      phoneNumber,
+      gender,
+      address,
+      email,
+      hashedPassword
+    );
+  } else {
+    console.log("[Warning!] - This email already exists!");
+  }
+};
+
+const signIn = async (email, password) => {
+  const user = await userDao.getUserByEmail(email);
+  const result = await bcrypt.compare(password, user.password);
+  if (result) {
+    const payLoad = {
+      exp: Math.floor(Date.now() / 1000) + 60 * 60 * 12 * 3,
+      id: user.id,
+    };
+    const secretKey = process.env.SECRETE_KEY;
+    const jwtToken = jwt.sign(payLoad, secretKey);
+    console.log(jwtToken);
+    return jwtToken;
+  }
+};
```

Comment:
```suggestion
const signIn = async (email, password) => {
  const user = await userDao.getUserByEmail(email);
  // 1
  if (!user) {
    throw new Error("NONEXISTENT_USER");
  }
  
  //2
  const isMatch = await bcrypt.compare(password, user.password);
  if (!isMatch) {
    throw new Error("INVALID_USER")
  }
  
  // 3
  const SSECRET_KEY = process.env.SECRETE_KEY;
  // 4
  const jwtToken = jwt.sign({ userId: user.id }, secretKey. {
      expireseIn: '9d'
  });
    
  return jwtToken
};
```

1. user 조회 후 user가 존재하지 않을 경우의 예외처리가 필요해보입니다!
2. 변수명을 좀 더 명확하게 수정 후, 비밀번호가 일치하지 않은 경우의 예외처리를 추가해주어야 합니다!
3. `secret key`는 상수 데이터로 변수 네이밍 컨벤션을 따라 변수명을 수정해주세요!
4. `token`의 페이로드 데이터로 유저의 정보와 토큰 발급 시간을 주입해주셨는데, 주입 시간은 토큰을 발급하는 그 시점에 자동으로 생성되며, 만약 만료기간을 고려하신 것이라면 3번째 인수로 받는 객체의 프로퍼티인 `expirseIn`에 대한 값을 설정해주면 됩니다. 공식문서를 확인하여 사용법을 확인해보고, 민감한 데이터라면 환경변수로 관리할 수 있을 것 같습니다~!